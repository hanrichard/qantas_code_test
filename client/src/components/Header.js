import React from 'react';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import Container from '@material-ui/core/Container';
import { Link } from 'react-router-dom';
import styled from 'styled-components';
import componentStyle from './Header.style';

const Wrapper = styled.div`
    ${componentStyle}
`;

const Header = () => {
    return (
        <Wrapper>
            <AppBar className="Header__appbar">
                <Toolbar>
                    <Container maxWidth="sm">
                        <Typography variant="h6">
                            <Link className="Header__link" to="/">
                                Qantas airport datas
                            </Link>
                        </Typography>
                    </Container>
                </Toolbar>
            </AppBar>
        </Wrapper>
    );
};

export default Header;
