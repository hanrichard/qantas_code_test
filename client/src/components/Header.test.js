import React from 'react';
import { shallow, configure } from 'enzyme';
import Header from './Header';
import Adapter from 'enzyme-adapter-react-16';
import Renderer from 'react-test-renderer';
configure({ adapter: new Adapter() });

function setup() {
    const shallowWrapper = shallow(<Header />);

    return {
        shallowWrapper,
    };
}

describe('Airport header', () => {
    const { shallowWrapper } = setup();
    it('should match snapshot', () => {
        const airportHeader = Renderer.create(`<BrowserRouter>shallowWrapper</BrowserRouter>`).toJSON();
        expect(airportHeader).toMatchSnapshot();
    });
});

describe('Header component', () => {
    it('should render', () => {
        const { shallowWrapper } = setup();
        expect(shallowWrapper.text().includes('Qantas airport datas')).toBe(true);
    });
});
