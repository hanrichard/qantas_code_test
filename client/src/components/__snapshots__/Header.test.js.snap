// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Airport header should match snapshot 1`] = `"<BrowserRouter>shallowWrapper</BrowserRouter>"`;
